services/system/Accounts/ui/src/components/AccountSelection.tsx

```diff
@@ -0,0 +1,722 @@
+        {
+          "bg-accent border ": isSelected,
+        }
+      )}
+    >
+      <div className="flex gap-3">
+        {!name ? (
+          <Skeleton className="h-12 w-12 rounded-full" />
+        ) : (
+          <Avatar>
+            <AvatarFallback>{name.toUpperCase().slice(0, 2)}</AvatarFallback>
+          </Avatar>
+        )}
+        <div className={cn("text-lg flex flex-col justify-center")}>
+          {!name ? (
+            <Skeleton className="h-4 w-[100px]" />
+          ) : (
+            capitaliseFirstLetter(name)
+          )}
+        </div>
+      </div>
+    </button>
+  );
+};
+
+interface AccountType {
+  account: string;
+  id: string;
+}
+
+const formSchema = z.object({
+  username: z.string().min(1).max(50),
+});
+
+const AccountNameStatus = z.enum(["Available", "Taken", "Invalid", "Loading"]);
+const GetAccountReturn = z
+  .object({
+    accountNum: z.string(),
+    authService: z.string(),
+    resourceBalance: z.boolean().or(z.bigint()),
+  })
+  .optional();
+
+const isAccountAvailable = async (
+  accountName: string
+): Promise<z.infer<typeof AccountNameStatus>> => {
+  try {
+    const res = GetAccountReturn.parse(
+      await supervisor.functionCall({
+        method: "getAccount",
+        params: [accountName],
+        service: "accounts",
+        intf: "api",
+      })
+    );
+
+    return AccountNameStatus.parse(res ? "Taken" : "Available");
+  } catch (e) {
+    console.error(e);
+    return AccountNameStatus.parse("Invalid");
+  }
+};
+
+export const AccountSelection = () => {
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      username: "",
+    },
+  });
+
+  const { mutateAsync: importAccount } = useMutation<void, string, string>({
+    mutationFn: async (accountName) => {
+      await supervisor.functionCall({
+        method: "importAccount",
+        params: [accountName],
+        service: "accounts",
+        intf: "admin",
+      });
+    },
+    onSuccess: async () => {
+      fetchAccounts();
+      setTimeout(() => {
+        fetchAccounts();
+      }, 2000);
+    },
+  });
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const [isCreatingAccount, setIsCreatingAccount] = useState(true);
+
+  const { isDirty, isSubmitting } = form.formState;
+
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    if (isCreatingAccount) {
+      void (await createAccount(values.username));
+    } else {
+      void (await importAccount(values.username));
+    }
+    setIsModalOpen(false);
+  };
+
+  const username = form.watch("username");
+
+  const [debouncedAccount, setDebouncedAccount] = useState<string>();
+
+  const {
+    data: accounts,
+    isLoading: isFetching,
+    error,
+    refetch: fetchAccounts,
+  } = useQuery({
+    queryKey: ["availableAccounts"],
+    queryFn: async (): Promise<AccountType[]> => {
+      supervisor.preLoadPlugins([{ service: "accounts" }]);
```

Comment:
Bit excessive to preload plugins every time you want to run this query. Preload only needs to happen once on page load